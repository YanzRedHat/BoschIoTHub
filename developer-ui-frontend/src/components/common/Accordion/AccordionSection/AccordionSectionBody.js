/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React, { Component } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Measure from "react-measure";
import { Motion, spring } from "react-motion";

const AnimationWrapper = styled.div`
  overflow: hidden;
`;

export default class AccordionSectionBody extends Component {
  constructor(props) {
    super(props);
    this.state = {
      height: -1
    };
  }

  render() {
    const { expanded, children, sticky } = this.props;
    const { height } = this.state;
    return !sticky ? (
      <Measure
        scroll
        onResize={contentRect => {
          this.setState({ height: contentRect.scroll.height });
        }}>
        {({ measureRef }) => (
          <Motion
            style={{
              height: spring(expanded ? height : 0, {
                stiffness: 300,
                damping: 27
              })
            }}
            defaultStyle={{ height: 0 }}>
            {interpolatingStyle => (
              <AnimationWrapper style={interpolatingStyle}>
                <div className="measuring-container" ref={measureRef}>
                  {children}
                </div>
              </AnimationWrapper>
            )}
          </Motion>
        )}
      </Measure>
    ) : (
      children
    );
  }
}

AccordionSectionBody.propTypes = {
  expanded: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired,
  sticky: PropTypes.bool
};
