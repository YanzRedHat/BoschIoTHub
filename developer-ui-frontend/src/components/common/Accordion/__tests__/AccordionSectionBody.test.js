/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import { shallow } from "enzyme";
import AccordionSectionBody from "components/common/Accordion/AccordionSection/AccordionSectionBody.js";

describe("AccordionSectionBody", () => {
  it("should be defined", () => {
    expect(AccordionSectionBody).toBeDefined();
  });
  it("should render correctly", () => {
    const tree = shallow(<AccordionSectionBody />);
    expect(tree).toMatchSnapshot();
  });
  it.skip("should have a prop scroll", () => {
    const tree = shallow(<AccordionSectionBody />);
    expect(tree.props().scroll).toEqual(true);
  });
  it.skip("should have a prop onResize", () => {
    const tree = shallow(<AccordionSectionBody />);
    expect(typeof tree.props().onResize).toBe("function");
  });
});
