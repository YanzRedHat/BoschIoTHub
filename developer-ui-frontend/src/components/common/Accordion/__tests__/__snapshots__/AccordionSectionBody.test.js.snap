// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionSectionBody should render correctly 1`] = `
<_class
  onResize={[Function]}
  scroll={true}
>
  <Component />
</_class>
`;
